### DISCUSSION TITLE
Bug: Can't delete attribute

### DISCUSSION THREAD

**Discussion by TribersNL** [swellstores/community#461] [Help] [answered]
Hi,

I can't delete an attribute eventhough there are no products attached to it. The error message tells me that it is a variant attribute, but it isn't.
<img width="1367" alt="Screenshot 2023-12-22 at 10 35 36" src="https://github.com/swellstores/community/assets/131139658/c2e165b7-df0a-4766-b2e6-084274c3e026">
<img width="1117" alt="Screenshot 2023-12-22 at 10 33 37" src="https://github.com/swellstores/community/assets/131139658/49bf99b4-594a-4cc2-8059-2ff2c6d02ac2">

**Comment by SusannaSwell** [comment]
Hello @TribersNL could you please specify your store id? for us to investigate

**Reply by studiodrimmel** [reply]
vivelamour is our storeid, sorry posted the original question with the wrong github account. It seems like I can't delete any attribute, at all.

**Reply by SusannaSwell** [reply, marked as answer]
no worries, thank you!
yes, I see that it is general issue with deleting attributes via admin UI. I will create a report in our internal system for fixing this case. will update you once I have the details.

As a workaround, you can delete attribute via backend API or developer console (open attribute, click on Actions, View in console, change request to DELETE)

**Reply by studiodrimmel** [reply]
Thanks, used that workaround for now. 

Unfortunately I have some more issues with attributes (and variants). Some of our products have only one option, like a golden necklace. But I can't filter on it, since the Color attribute is a variant type and I can't use it without making a product with one single variant :/ But since it is a simple product, I would assume that just adding the attribute Color or add a product Option without creating Variants would do the trick.

Is this intentionally? If so, I would need to think of a workaround.

**Reply by SusannaSwell** [reply]
It is not possible to associate Variant type of attribute with simple products (without variants). It is applied automatically when you generate variants from an option.

Other way, if you already have Color attribute with variant type and you do not want to generate single variant for the product with one option.  You can create an attribute with another name (Colors, etc.) with Checkbox type make it filterable, add colors that you may need for simple products (or add all colors), then apply created attribute to product with choosing available color.

**Comment by SusannaSwell** [comment]
@studiodrimmel  the issue with deleting attribute is fixed now. You should be able to delete a standard attribute with no products attached to it.
